### ISSUE TITLE
[no-unused-vars] False positive in abstract class method definitions

### ISSUE THREAD

**Issue by mgummelt** [typescript-eslint/typescript-eslint#3759]
- [ X] I have tried restarting my IDE and the issue persists.
- [ X] I have updated to the latest version of the packages.
- [ X] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "no-unused-vars": "off",
    "@typescript-eslint/no-unused-vars": "error"
  }
}
```

```TS
export default abstract class AbstractType {
    apiValueToRenderable(
        value: APIValue | undefined,
        rowID: string): Renderable {

        if (isObjectPrimitive(value)) {
            return JSON.stringify(value);
        } else {
            return value;
        }
    }
}
```

**Expected Result**

No lint errors.

The body of the abstract class doesn't use `rowID`, but if I remove it, the child classes that do use `rowID` won't type check, because they'll be using a parameter that doesn't exist in the parent method definition.

**Actual Result**

`'rowID' is defined but never used.`

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.29.2` |
| `@typescript-eslint/parser`        | `4.29.2` |
| `TypeScript`                       | `4.3.5` |
| `ESLint`                           | `7.32.0` |
| `node`                             | `14.17.5` |

**Comment by bradzacher** [comment]
This is not a false positive - this is working as intended.
You have a clearly unused variable that you are leaving in the code *on purpose*.

This is the exact reason that the disable comment was created for.
Add a disable comment and for good measure add another comment explaining to future readers why you have an obviously unused variable in the code.
